charts/retool/templates/telemetry_deployment.yaml

```diff
@@ -0,0 +1,158 @@
+{{- if .Values.telemetry.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "retool.fullname" . }}-telemetry
+  labels:
+    {{- include "retool.telemetry.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "retool.telemetry.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "retool.telemetry.selectorLabels" . | nindent 8 }}
+      annotations:
+        {{- if or .Values.telemetry.customVectorConfig .Values.telemetry.customGrafanaAgentConfig }}
+        checksum/custom-config: {{ include (print $.Template.BasePath "/telemetry_configmap.yaml") . | sha256sum | quote }}
+        {{- end }}
```

Comment:
this makes it so that custom config changes will still roll the telemetry pod even if the pod itself doesn't change